src/database/database.ts

```diff
@@ -1,233 +1,75 @@
-
-  constructor(app: FirebaseApp, dbUrl: string) {
-    const parsedUrl = new URL(dbUrl);
-    parsedUrl.pathname = path.join(parsedUrl.pathname, RULES_URL_PATH);
-    this.dbUrl = parsedUrl.toString();
-    this.httpClient = new AuthorizedHttpClient(app);
-  }
-
-  /**
-   * Gets the currently applied security rules as a string. The return value consists of
-   * the rules source including comments.
-   *
-   * @return {Promise<string>} A promise fulfilled with the rules as a raw string.
-   */
-  public getRules(): Promise<string> {
-    const req: HttpRequestConfig = {
-      method: 'GET',
-      url: this.dbUrl,
-    };
-    return this.httpClient.send(req)
-      .then((resp) => {
-        if (!resp.text) {
-          throw new FirebaseAppError(AppErrorCodes.INTERNAL_ERROR, 'HTTP response missing data.');
-        }
-        return resp.text;
-      })
-      .catch((err) => {
-        throw this.handleError(err);
-      });
-  }
-
-  /**
-   * Gets the currently applied security rules as a parsed JSON object. Any comments in
-   * the original source are stripped away.
-   *
-   * @return {Promise<object>} A promise fulfilled with the parsed rules source.
-   */
-  public getRulesJSON(): Promise<object> {
-    const req: HttpRequestConfig = {
-      method: 'GET',
-      url: this.dbUrl,
-      data: { format: 'strict' },
-    };
-    return this.httpClient.send(req)
-      .then((resp) => {
-        return resp.data;
-      })
-      .catch((err) => {
-        throw this.handleError(err);
-      });
-  }
-
-  /**
-   * Sets the specified rules on the Firebase Database instance. If the rules source is
-   * specified as a string or a Buffer, it may include comments.
-   *
-   * @param {string|Buffer|object} source Source of the rules to apply. Must not be `null`
-   *  or empty.
-   * @return {Promise<void>} Resolves when the rules are set on the Database.
-   */
-  public setRules(source: string | Buffer | object): Promise<void> {
-    if (!validator.isNonEmptyString(source) &&
-      !validator.isBuffer(source)  &&
-      !validator.isNonNullObject(source)) {
-      const error = new FirebaseDatabaseError({
-        code: 'invalid-argument',
-        message: 'Source must be a non-empty string, Buffer or an object.',
-      });
-      return Promise.reject(error);
-    }
-
-    const req: HttpRequestConfig = {
-      method: 'PUT',
-      url: this.dbUrl,
-      data: source,
-      headers: {
-        'content-type': 'application/json; charset=utf-8',
-      },
-    };
-    return this.httpClient.send(req)
-      .then(() => {
-        return;
-      })
-      .catch((err) => {
-        throw this.handleError(err);
-      });
-  }
 
-  private handleError(err: Error): Error {
-    if (err instanceof HttpError) {
-      return new FirebaseDatabaseError({
-        code: AppErrorCodes.INTERNAL_ERROR,
-        message: this.getErrorMessage(err),
-      });
-    }
-    return err;
-  }
+declare module '@firebase/database-types' {
```

Comment:
See my other comment about loading everything from `@firebase/database`. In that case this won't be needed.